compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/window/window/WindowStateTest.kt

```diff
@@ -310,6 +310,130 @@ class WindowStateTest {
         assertThat(window.placement).isEqualTo(WindowPlacement.Floating)
     }
 
+    @Test
+    fun `from floating to fullscreen to floating`() = runApplicationTest(
+        useDelay = isLinux || isMacOs,
+        delayMillis = 1000
+    ) {
+        val state = WindowState(size = DpSize(200.dp, 200.dp))
+        lateinit var window: ComposeWindow
+
+        launchTestApplication {
+            Window(onCloseRequest = {}, state) {
+                window = this.window
+            }
+        }
+
+        awaitIdle()
+
+        // Enter fullscreen from floating
+        state.placement = WindowPlacement.Floating
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Floating)
+
+        state.placement = WindowPlacement.Fullscreen
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Fullscreen)
+
+        // Exit fullscreen to floating
+        state.placement = WindowPlacement.Floating
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Floating)
+    }
+
+    @Test
+    fun `from maximized to fullscreen to floating`() = runApplicationTest(
+        useDelay = isLinux || isMacOs,
+        delayMillis = 1000
+    ) {
+        val state = WindowState(size = DpSize(200.dp, 200.dp))
+        lateinit var window: ComposeWindow
+
+        launchTestApplication {
+            Window(onCloseRequest = {}, state) {
+                window = this.window
+            }
+        }
+
+        awaitIdle()
+
+        // Enter fullscreen from maximized
+        state.placement = WindowPlacement.Maximized
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Maximized)
+
+        state.placement = WindowPlacement.Fullscreen
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Fullscreen)
+
+        // Exit fullscreen to floating
+        state.placement = WindowPlacement.Floating
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Floating)
+    }
+
+    @Test
+    fun `from maximized to fullscreen to maximized`() = runApplicationTest(
+        useDelay = isLinux || isMacOs,
+        delayMillis = 1000
+    ) {
+        val state = WindowState(size = DpSize(200.dp, 200.dp))
+        lateinit var window: ComposeWindow
+
+        launchTestApplication {
+            Window(onCloseRequest = {}, state) {
+                window = this.window
+            }
+        }
+
+        awaitIdle()
+
+        // Enter fullscreen from maximized
+        state.placement = WindowPlacement.Maximized
+        awaitIdle()
+        assertThat(window.placement).isEqualTo(WindowPlacement.Maximized)
```

Comment:
Tests [fail](https://teamcity.jetbrains.com/buildConfiguration/JetBrainsPublicProjects_Compose_DesktopTestsForPullRequests/4508579?buildTab=overview&expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&expandBuildTestsSection=true&hideProblemsFromDependencies=false) on Linux in these lines:
```
expected: Maximized
but was : Floating
```